dlib/pixel.h

```diff
@@ -771,6 +865,158 @@ namespace dlib
             return(c2);
         }
 
+        // -----------------------------
+
+        struct Lab
+        {
+            double l;
+            double a;
+            double b;
+        };
+        /*
+            Calculate Lab from RGB
+            L is between 0 and 100
```

Comment:
Ok, I see how it is done with the hsi_pixel. The representation is correct, it's just shifted and scaled to fit "unsigned char". So when accessing the members directly, you have to do the shift and scale first. 

What I'm trying to say, is that you cannot assign values to the lab_pixel using the struct members directly, but you cannot do that with hsi_pixel either:
`//RGB #FF00AA (magenta)
rgb_pixel magenta();
magenta.r = 0xff;
magenta.g = 0x0;
magenta.b = 0xaa;

// HSL[320, 100%, 50%] (same magenta)
hsi_pixel magenta();
magenta.h = 320; // not possible (use 320/360 * 256)
magenta.s = 100; // incorrect
magenta.i = 50; // incorrect
`
So I will do it in same way as the HSI works.